lib/galaxy/workflow/modules.py

```diff
@@ -1196,7 +1237,7 @@ def populate_module_and_state(trans, workflow, param_map, allow_tool_state_corre
         module_injector = WorkflowModuleInjector(trans, allow_tool_state_corrections)
     for step in workflow.steps:
         step_args = param_map.get(step.id, {})
-        step_errors = module_injector.inject(step, step_args=step_args)
+        step_errors = module_injector.inject(step, step_args=step_args, param_map=step_args)
```

Comment:
Passing step_args twice looks a bit funny here, would it be problematic to always use `step_args` on line 1215 ?

Reply:
Let me see how else it is used - but yeah that is definitely a good idea.

Reply:
So ``WorkflowProgress.remaining_steps()`` called inject without step_args previously - so making this change will change that - so this line will change:

```
        state, step_errors = module.compute_runtime_state(self.trans, step_args)
        step.state = state
```

I don't know if that will introduce bugs or fix bugs - but probably one of them. We will see what the tests say - but the UI exercises this API in ways that the tests don't cover. The new test ``test_run_runtime_parameters_after_pause`` still passes with this change - I guess that seems like the kind of thing most likely to break.